include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -183,7 +256,7 @@ class ExtendedThreadPoolInterface : public Eigen::ThreadPoolInterface {
   // [0,k) where k<=n.
   virtual void RunInParallelSection(ThreadPoolParallelSection &ps,
                                     std::function<void(unsigned idx)> fn,
-                                    unsigned n) = 0;
+                                    unsigned n, std::ptrdiff_t block_size) = 0;
```

Comment:
Could we call this debug_block_size, to be clearer that it is passed in just to make it available to enable logging.  Alternatively, can we generate this per-loop logging from within threadpool.cc where the block size is computed?

Reply:
Prefer "debug_block_size" approach - block size is calculated inside thread pool while the profiler instance is owned by thread pool impl. Will have to add more logic to bridge the gap were to let the thread pool log the size itself.